tls/handshake_client.go

```diff
@@ -330,6 +330,13 @@ func (c *Conn) clientHandshake() error {
 			secureRenegotiation:  true,
 			alpnProtocols:        c.config.NextProtos,
 			extendedMasterSecret: c.config.maxVersion() >= VersionTLS10 && c.config.ExtendedMasterSecret,
+			// supportedVersion is order of preference. TODO: VersionTLS13
+			// This is used for the Version extension (RFC 8446 Sec 4.2.1)
+			supportedVersions: []uint16{
```

Comment:
This should be set based on the values in `config.MaxVersion` and `config.MinVersion`.